File: OrganizationService/src/main/java/com/slavacom/organizationservice/dto/old/ProjectTagsDto.java
package com.slavacom.organizationservice.dto.old;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.slavacom.organizationservice.entity.ProjectTags;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

/**
 * DTO for {@link ProjectTags}
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class ProjectTagsDto {
	private UUID id;
	private UUID projectId;
	@NotNull
	@NotEmpty
	@NotBlank
	private String name;
	private String color;
	private String description;
	private Instant createdAt;
	private Instant updatedAt;
}

File: OrganizationService/src/main/java/com/slavacom/organizationservice/project/tags/ProjectTagsDto.kt
package com.slavacom.organizationservice.project.tags

import java.time.Instant
import java.util.UUID

data class ProjectTagResponse(
    val id: UUID,
    val projectId: UUID,
    val organizationId: UUID?,
    val name: String,
    val color: String?,
    val description: String?,
    val createdAt: Instant,
    val updatedAt: Instant?
)

data class CreateProjectTagRequest(
    val name: String,
    val color: String? = null,
    val description: String? = null
)

data class UpdateProjectTagRequest(
    val name: String? = null,
    val color: String? = null,
    val description: String? = null
)
